enhancements/cluster-logging/cluster-logging-log-forwarding.md

```diff
@@ -0,0 +1,320 @@
+---
+title: cluster-logging-log-forwarding
+authors:
+  - "@jcantrill"
+reviewers:
+  - "@bparees"
+  - "@ewolinetz"
+  - "@jeremyeder"
+approvers:
+  - "@bparees"
+  - "@ewolinetz"
+  - "@richm"
+creation-date: 2019-09-17
+last-updated: 2019-09-17
+status: implementable
+see-also:[]
+replaces:[]
+superseded-by:[]
+---
+
+# cluster-logging-log-forwarding
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Summary
+
+The purpose of log forwarding is to provide a declarative way by which adopters of cluster logging can ship
+container and node logs to destinations that are not necessarily managed by the OKD cluster logging infrastructure.
+Destinations are either on or off cluster endpoints such as the cluster logging provided Elasticsearch, an organization's Kafka message bus, or other log aggregation solutions (e.g. Splunk, Graylog).
+
+## Motivation
+
+Organizations desire to reuse their existing enterprise log solutions to store container logs.  Providing a declarative mechanism by which administrators define a log destination simplifies their operational burden.  They are able to take advantage of log collection infrastructure with minimal configuration changes. The cluster logging stack is able to deploy a collector to each OKD node that is configured with the necessary permissions to collect the logs, add container metadata (e.g. labels) and ship them to the specified endpoint.
+
+
+
+### Goals
+The specific goals of this proposal are:
+
+* Provide an API that identifies log sources and their desired outputs.
+* Send logs to an Elasticsearch cluster not managed by the cluster logging infrastructure
+* Send logs to a fluentd instance not managed by the cluster logging infrastructure via fluentd's secureforward protocol
+* Allow deployment of log forwarding without deploying the entirety of the cluster logging infrastructure (e.g. Kibana, Elasticsearch)
```

Comment:
not seeing how the ClusteringLogging CRD enables this decision/configuration?  is this still a goal of this design?

Reply:
currently, you would basically not configure a `logStore` or `visualization` afaik.

Reply:
if that's already possible today, i don't know why it's described in the EP?

if we're proposing changing how it works(or enabling it because it's not possible today), then this EP needs more detail on that.

Reply:
@bparees should i clarify it:  its possible today but it means:
* admin makes stack unmanaged
* admin fixes up the fluent config manually

This proposal would move the 'management' of forwarding to the operator and CR without needing to modify the config though admins will need to still setup secrets and the servie

Reply:
what i'm missing is how this allows me to avoid deploying kibana and elastic, as is implied by the statement `Allow deployment of log forwarding without deploying the entirety of the cluster logging infrastructure (e.g. Kibana, Elasticsearch)`

I get that this EP allows an admin to setup log forwarding w/o going "unmanaged", but it does not allow them to "disable" the other parts of the stack, from what I see.  If it does allow them to disable other parts of the stack, i need to understand how that is accomplished.  If they only way to do it involves going "unmanaged" then effectively it can't be done and we shouldn't talk about it.